### PR TITLE
test: Added test of public data model view schemas

### PR DESCRIPTION
**Pull request by energinet-ajw**
Each view must conform to a schema to verify correctness (types, names and order).
Nullablity is not tested.

- [x] The title adheres to [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [x] Tests are written and executed locally
- [ ] Subsystem tests have been tested (by manually deploying to `dev_002`)

### CODE REVIEW

File: source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/calculation_results/amount_per_charge_v1_schema.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        StructField("calculation_type", StringType(), False),
+        StructField("calculation_version", LongType(), False),
+        StructField("result_id", StringType(), False),
+        StructField("grid_area_code", StringType(), False),
+        StructField("energy_supplier_id", StringType(), False),
+        StructField("charge_code", StringType(), False),
+        StructField("charge_type", StringType(), False),
+        StructField("charge_owner_id", StringType(), False),
+        StructField("resolution", StringType(), False),
+        StructField("quantity_unit", StringType(), True),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/calculation_results/amount_per_charge_v1_schema.py:38]: Isn't this False?

Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/calculation_results/amount_per_charge_v1_schema.py:38]: Do you mean quantity_unit?

Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/calculation_results/amount_per_charge_v1_schema.py:38]: Yes to quantity_unit. I will change it.

File: source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/charge_link_periods_v1_schema.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        StructField("metering_point_id", StringType(), False),
+        StructField("metering_point_type", StringType(), False),
+        StructField("charge_type", StringType(), False),
+        StructField("charge_code", StringType(), False),
+        StructField("charge_owner_id", StringType(), False),
+        StructField("charge_link_quantity", IntegerType(), False),
+        StructField("from_date", TimestampType(), False),
+        StructField("to_date", TimestampType(), True),
+        StructField("grid_area_code", StringType(), False),
+        StructField("energy_supplier_id", StringType(), True),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/charge_link_periods_v1_schema.py:38]: I think this is False

Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/charge_link_periods_v1_schema.py:38]: basis_data.metering_point_types -> energy_supplier_id STRING,

File: source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/monthly_amounts_v1_schema.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        StructField("calculation_version", LongType(), False),
+        StructField("result_id", StringType(), False),
+        StructField("grid_area_code", StringType(), False),
+        StructField("energy_supplier_id", StringType(), False),
+        StructField("time", TimestampType(), False),
+        StructField("resolution", StringType(), False),
+        StructField("quantity_unit", StringType(), False),
+        StructField("currency", StringType(), False),
+        StructField("amount", DecimalType(18, 6), True),
+        StructField("charge_type", StringType(), True),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/monthly_amounts_v1_schema.py:38]: False

File: source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/monthly_amounts_v1_schema.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        StructField("charge_code", StringType(), True),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/monthly_amounts_v1_schema.py:39]: False

File: source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/monthly_amounts_v1_schema.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        StructField("charge_owner_id", StringType(), True),
```
Comment (JesperGomes) [inline source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/monthly_amounts_v1_schema.py:40]: False

File: source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/wholesale_results_v1_schema.py [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        StructField("time", TimestampType(), False),
+        StructField("resolution", StringType(), False),
+        StructField("metering_point_type", StringType(), False),
+        StructField("settlement_method", StringType(), True),
+        StructField("quantity_unit", StringType(), False),
+        StructField("currency", StringType(), False),
+        StructField("quantity", DecimalType(18, 3), False),
+        StructField("price", DecimalType(18, 6), False),
+        StructField("amount", DecimalType(18, 6), True),
+        StructField("charge_type", StringType(), True),
```
[the remark this answers is not in the document]
Comment (energinet-ajw) [reply source/databricks/calculation_engine/tests/features/utils/views/expected_schemas/settlement_report/wholesale_results_v1_schema.py:42]: Task made.

**Review by JesperGomes — APPROVED** [review]

**Pull request merged**